src/TokenAirdrop.hs

```diff
@@ -43,10 +51,30 @@ tokenAirdrop config = do
         utxos <- utxosAt config $ config.ownAddress
         let lookups = Constraints.unspentOutputs utxos
 
-        submitTx @Void config pubKeyAddressMap lookups tx
+        eTxId <- submitTx @Void config pubKeyAddressMap lookups tx
+        case eTxId of
+          Left err -> pure $ Left err
+          Right txId -> do
+            putStrLn $ "Submitted transaction successfully: " ++ show txId
+            putStrLn "Waiting for confirmation..."
+            waitUntilHasTxIn config 0 txId
+            pure $ Right ()
     )
     $ zipWith combine2To3 txPairs [1 :: Int ..]
 
+-- | Repeatedly waits a block until we have the inputs we need
+waitUntilHasTxIn :: Config -> Integer -> TxId -> IO ()
+waitUntilHasTxIn config n txId = do
+  when (n > blockCountWarning) $ putStrLn "WARNING: Waited over 50 blocks for transaction to land, it may have failed. (Continuing to wait)"
```

Comment:
I.e.
```
WARNING: Waited 51 blocks for transaction to land, it may have failed. (Continuing to wait)
WARNING: Waited 52 blocks for transaction to land, it may have failed. (Continuing to wait)
```

Reply:
Good point, I'll fix that